### PR TITLE
Add support to define the alignment of structs

### PR DESCRIPTION
**Pull request by henriquemotaesteves**
This pull request is related to the issue #8.

This pull request was tested in the following platforms:

OS | Arch
------------ | -------------
Windows 7 | X86 - 32-bit
Windows 7 | X86 - 64-bit
Ubuntu 14.04 | X86 - 32-bit
Ubuntu 14.04 | X86 - 64-bit
Max OS X 10.10.2 | X86 - 64-bit

### CODE REVIEW

**Comment by henriqueesteves** [comment]
Do you have any feedback?

File: src/main/java/jnr/ffi/Struct.java [hunk 1]
```diff
@@ -239,6 +245,34 @@ public double doubleValue() {
         }
     }
 
+    public static final class Alignment extends Number {
```
Comment (headius) [inline src/main/java/jnr/ffi/Struct.java]: I am a bit confused why this type is needed. Isn't this essentially just java.lang.Integer? Will we not want to support long alignments as well?

Comment (henriquemotaesteves) [inline src/main/java/jnr/ffi/Struct.java]: This is just a tiny type and it was based on the Offset type. I just followed the same design. But if you prefer I can replace it by an 'int' or 'long' without problems.

https://github.com/jnr/jnr-ffi/blob/master/src/main/java/jnr/ffi/Struct.java#L224

**Comment by goto1134** [comment]
Don't forget to change the Padding inner class to align correctly.

File: src/main/java/jnr/ffi/Struct.java (hunk 1 above)
Comment (headius) [inline src/main/java/jnr/ffi/Struct.java]: Ahh, I see. I wonder if this was an attempt at improving type-safety. I don't really see the value in these Number subtypes that just wrap int, but we'll leave it for now.

**Comment by headius** [comment]
@goto1134 Can you express that in the form of a patch? :-) I am not sure what needs to change.

File: src/main/java/jnr/ffi/Struct.java [hunk 2]
```diff
@@ -101,13 +103,12 @@ protected final int addField(int sizeBits, int alignBits, Offset offset) {
         }
 
         protected final int addField(int sizeBits, int alignBits) {
-            final int off = resetIndex ? 0 : align(this.size, alignBits >> 3);
-            this.size = Math.max(this.size, off + (sizeBits >> 3));
-            this.minAlign = Math.max(this.minAlign, alignBits >> 3);
-            return off;
+            final int alignment = this.alignment.intValue() > 0 ? Math.min(this.alignment.intValue(), (alignBits >> 3)) : (alignBits >> 3);
```
Comment (goto1134) [inline src/main/java/jnr/ffi/Struct.java]: What will be if (alignBits >> 3) are less than alignment parameter? The field will align with less padding value? As I see, if the alignment parameter is set, it must be applied to any field.

Comment (goto1134) [inline src/main/java/jnr/ffi/Struct.java]: Try a Test when you put two unsigned8 field in a struct with padding set to 16. The first must have 0 offset, the second must have 16.

Comment (henriquemotaesteves) [inline src/main/java/jnr/ffi/Struct.java]: @goto1134 is not this test case already tested in this [test case](https://github.com/jnr/jnr-ffi/pull/46/files#diff-bd5234f278cf4e3ece9bfc282ac64224R164)?

Comment (headius) [inline src/main/java/jnr/ffi/Struct.java]: @goto1134 Does #46 look like it addresses your concern?

Comment (goto1134) [inline src/main/java/jnr/ffi/Struct.java]: @henriquemotaesteves ah, now I see that I just misunderstood the way it works.

Comment (headius) [inline src/main/java/jnr/ffi/Struct.java]: @goto1134 Does this mean that #82 is no longer valid? The fixes here do not address your test cases (i.e. they still fail).

Comment (goto1134) [inline src/main/java/jnr/ffi/Struct.java]: @headius This means that my comment about the size of the structs were incorrect.

The problem #82 has no connection with this issue. It is still valid. 

btw, sorry for confusion.

**Pull request merged**